libutils/buffer.c

```diff
@@ -374,7 +375,11 @@ int BufferPrintf(Buffer *buffer, const char *format, ...)
      * everything is easy, however if we are shared then we need a different strategy.
      */
     int printed = vsnprintf(buffer->buffer, buffer->capacity, format, aq);
-    if (printed >= buffer->capacity)
+    if (printed < 0) 
+    {
+        // vsnprintf failed!
```

Comment:
@larsewi @olehermanse @vpodzime I wonder about these comments? Should it be a TODO to do something about the case where vsnprintf fails?

Reply:
@craigcomstock in this case `BufferPrintf` returns the negative value of `vsnprintf` when it failes. Thus the caller of `BufferPrintf` must handle it. I don't think it makes sence to handle such errors in a utillity function.